05/app.js

```diff
@@ -0,0 +1,41 @@
+function Student(name, lastname){
+    this.firstName = name;
+    this.lastName = lastname;
+    this.grades = {};
+}
+
+Student.prototype.addGrade = function(subject, grade){
+    if(typeof this.grades[subject] === 'undefined'){
+    this.grades[subject] = [];
+    };
+    this.grades[subject].push(grade);
+}
+
+Student.prototype.getAverageGrade = function(subject) {
+    // suma ocen
+    let result = 0;
+    if(subject) {
```

Comment:
Warto sprawdzić czy jest to `undefined` tj. `typeof subject === 'undefined'` - wtedy mamy pewność, że ten parametr nie został przekazany